projects/core/src/cms/page/base-page-meta.resolver.ts

```diff
@@ -1,25 +1,35 @@
 import { Injectable } from '@angular/core';
-import { combineLatest, defer, Observable } from 'rxjs';
-import { filter, map, shareReplay } from 'rxjs/operators';
+import { NavigationEnd, Router } from '@angular/router';
+import { combineLatest, defer, Observable, of } from 'rxjs';
+import { filter, map, shareReplay, startWith } from 'rxjs/operators';
 import { TranslationService } from '../../i18n/translation.service';
 import { CmsService } from '../facade/cms.service';
 import { BreadcrumbMeta, Page, PageRobotsMeta } from '../model/page.model';
+import { CanonicalUrlOptions } from './config/page-meta.config';
 import {
+  CanonicalPageResolver,
   PageBreadcrumbResolver,
   PageRobotsResolver,
   PageTitleResolver,
 } from './page.resolvers';
+import { PageLinkService } from './routing/page-link.service';
 import { RoutingPageMetaResolver } from './routing/routing-page-meta.resolver';
 
 @Injectable({
   providedIn: 'root',
 })
 export class BasePageMetaResolver
-  implements PageTitleResolver, PageBreadcrumbResolver, PageRobotsResolver {
+  implements
+    PageTitleResolver,
+    PageBreadcrumbResolver,
+    PageRobotsResolver,
+    CanonicalPageResolver {
   constructor(
     protected cmsService: CmsService,
     protected translation: TranslationService,
-    protected routingPageMetaResolver: RoutingPageMetaResolver
+    protected routingPageMetaResolver: RoutingPageMetaResolver,
+    protected router?: Router,
+    protected pageLinkFactory?: PageLinkService
```

Comment:
[nitpick] We've already renamed the factory to service for the class, but left the references `pageLinkFactory` (here and in other places).